File: 27-remove-element.go
func removeElement(nums []int, val int) int {
    if len(nums) == 0 {
        return 0
    }
    st, ed := 0, 0
    for ed < len(nums){
        if nums[ed] != val {
            nums[st], nums[ed] = nums[ed], nums[st]
            st += 1
        }
        ed += 1
    }
	fmt.Println(nums)
    return st
}

File: bwu/array/27-remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        p, n = 0, len(nums)
        for i in xrange(n):
            if nums[i] != val:
                nums[p] = nums[i]
                p += 1
        return p
